helpers/location_service.py

```diff
@@ -1,166 +1,240 @@
+        if config.data["sharing"]["discord_channel"]:
+            key = config.data["discord"]["url"].split("?")[0].split("/")[-1]
+            if key != "" and self.group_key != key:
+                self.group_key = key
+        else:
+            if self.group_key != config.data["sharing"]["group_key"]:
+                self.group_key = config.data["sharing"]["group_key"]
+        
+        # Set self.host from config
+        self.host = config.data["sharing"]["url"]
+
+        # Set self.zone_name from config
+        self.zone_name = config.data["maps"]["last_zone"]
+
+        # Start the service from init if sharing is enabled in the config
+        if config.data.get("sharing", {}).get("enabled", False):
+            self.start()
+
+    # Used to start the websocket service and connect signals
+    def start(self):
+        # Setup Websocket
+        self.websocket = QWebSocket()
+        self.websocket.connected.connect(self.connected)
+        self.websocket.error.connect(self.error)
+        self.websocket.disconnected.connect(self.disconnected)
+        self.websocket.textMessageReceived.connect(self.message)
+        self.websocket.open(QUrl(self.host))
+
+        # Setup signals
+        QApplication.instance().aboutToQuit.connect(self.stop)
+        QApplication.instance()._signals["logreader"].character_updated.connect(self.character_updated)
+        QApplication.instance()._signals["maps"].new_zone.connect(self.zone_updated)
+        QApplication.instance()._signals["maps"].location.connect(self.share_location)
+        QApplication.instance()._signals["maps"].death.connect(self.share_death)
+        QApplication.instance()._signals["locationsharing"].textMessageReceived.connect(self.parse)
+
+        # Set self.running to true
+        self.running = True
+
+    # Used to stop the websocket service and disconnect signals
+    def stop(self):
+        # Shutdown the websocket if it exists
+        if self.websocket:
+            self.websocket.close()
+            self.websocket = None
+
+        # Disconnect signals
+        QApplication.instance().aboutToQuit.disconnect(self.stop)
+        QApplication.instance()._signals["logreader"].character_updated.disconnect(self.character_updated)
+        QApplication.instance()._signals["maps"].new_zone.disconnect(self.zone_updated)
+        QApplication.instance()._signals["maps"].location.disconnect(self.share_location)
+        QApplication.instance()._signals["maps"].death.disconnect(self.share_death)
+        QApplication.instance()._signals["locationsharing"].textMessageReceived.disconnect(self.parse)
+
+        # Set self.running to false
+        self.running = False
+
+    def config_updated(self):
+        # Handle performing character name override when saving a new name from settings
+        if config.data["sharing"]["enabled"] and config.data["sharing"]["player_name_override"] and config.data["sharing"]["player_name"] != self.character_name:
```

Comment:
I think if sharing is disabled we can quickly short circuit (and call the stop from line 110-112), but even if we keep going I think it's safe to update the sharing name regardless of whether it's enabled (and this is the only one you're bothering to check).

Long winded way of saying you could probably remove `config.data["sharing"]["enabled"] and` from the check above